File: src/Application/Models/TravelPlan.cs
using Application.Models.Flights;

namespace Application.Models;

public class TravelPlan
{
    public string? Origin { get; set; }
    public string? Destination { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }

    public TravelPlanStatus TravelPlanStatus { get; set; } = TravelPlanStatus.NotStarted;

    public FlightPlan FlightPlan { get; init; } = new();

    public void AddFlightSearchOption(FlightOptionSearch flightOptions)
    {
        FlightPlan.AddFlightOptions(flightOptions);
    }

    public void SelectFlightOption(FlightOption flightOption)
    {
        FlightPlan.SelectFlightOption(flightOption);

        TravelPlanStatus = TravelPlanStatus.Completed;
    }
}

File: ui/src/features/travel-planning/domain/TravelPlan.ts
export interface TravelPlan {
    id: string;
    destination: string;
    startDate: Date | null;
    endDate: Date | null;
    origin: string;
    flightPlan?: FlightPlan;
}

export interface FlightPlan {
    flightOption?: FlightOption | null;
}

export interface FlightOption {
    airline: string;
    flightNumber: string;
    departure: FlightEndpoint;
    arrival: FlightEndpoint;
    duration: string;
    price: Price;
}

export interface FlightEndpoint {
    airport: string;
    airportCode: string;
    datetime: Date | null;
}

export interface Price {
    amount: number;
    currency: string;
}